docs/topics/control-flow.md

```diff
@@ -177,6 +177,82 @@ fun Request.getBody() =
 
 The scope of variable introduced in *when* subject is restricted to the body of this *when*.
 
+### Guard conditions in when expressions
+
+> Guard conditions are an [experimental feature](components-stability.md#stability-levels-explained) that may be changed at any time.
+> We would appreciate your feedback in [YouTrack](https://youtrack.jetbrains.com/issue/KT-71140/Guard-conditions-in-when-expressions-feedback).
+>
+{type="warning"}
+
+Starting from Kotlin 2.1, you can use guard conditions in `when` expressions or statements with a subject.
```

Comment:
Actually, we try to avoid such phrases since our documentation reflects the latest state of the language.